src/components/refactor/dialogs/generator/set-points/set-points-utils.js

```diff
@@ -0,0 +1,196 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import {
+    ACTIVE_POWER_SET_POINT,
+    DROOP,
+    EQUIPMENT,
+    FREQUENCY_REGULATION,
+    ID,
+    NAME,
+    NOMINAL_VOLTAGE,
+    Q_PERCENT,
+    REACTIVE_POWER_SET_POINT,
+    SUBSTATION_ID,
+    TOPOLOGY_KIND,
+    TYPE,
+    VOLTAGE_LEVEL,
+    VOLTAGE_REGULATION,
+    VOLTAGE_REGULATION_TYPE,
+    VOLTAGE_SET_POINT,
+} from '../../../utils/field-constants';
+import yup from '../../../utils/yup-config';
+import { REGULATION_TYPES } from '../../../../network/constants';
+import { getRegulatingTerminalEmptyFormData } from '../../regulating-terminal/regulating-terminal-form-utils';
+import {
+    PREVIOUS_ACTIVE_POWER_SET_POINT,
+    PREVIOUS_DROOP,
+    PREVIOUS_EQUIPMENT,
+    PREVIOUS_FREQUENCY_REGULATION,
+    PREVIOUS_VOLTAGE_LEVEL,
+    PREVIOUS_VOLTAGE_REGULATION,
+    PREVIOUS_VOLTAGE_SET_POINT,
+} from '../modification/generator-modification-utils';
+
+const getFrequencyRegulationEmptyFormData = (frequencyRegulation) => ({
+    [FREQUENCY_REGULATION]: frequencyRegulation,
+    [DROOP]: null,
+});
+
+const getFrequencyRegulationSchema = () => ({
+    [FREQUENCY_REGULATION]: yup
+        .bool()
+        .nullable()
+        .when([PREVIOUS_FREQUENCY_REGULATION], {
+            is: (value) => value == null,
+            then: (schema) => schema.required(),
+        }),
+    [DROOP]: yup
+        .number()
+        .nullable()
+        .when([FREQUENCY_REGULATION, PREVIOUS_DROOP], {
+            is: (frequencyRegulation, previousDroop) =>
+                frequencyRegulation && previousDroop == null,
+            then: (schema) => schema.required(),
+        }),
+});
+
+const getVoltageRegulationEmptyFormData = (voltageRegulationType) => ({
+    [VOLTAGE_SET_POINT]: null,
+    [Q_PERCENT]: null,
+    [VOLTAGE_REGULATION_TYPE]: voltageRegulationType,
+    [VOLTAGE_LEVEL]: null,
+    [EQUIPMENT]: null,
```

Comment:
Remove [EQUIPMENT] and [VOLTAGE_LEVEL] as their values are being set in getRegulatingTerminalEmptyFormData()